fix(jsonls): preserve already configured json schemas

okuuva opened the pull request.

mrjones2014 commented:
Would you mind adding a unit test for it? Otherwise I can add one in a follow up PR.

mrjones2014 commented on spec/jsonls_spec.lua:
```diff
@@ -0,0 +1,222 @@
+---@module 'luassert'
+
+describe('jsonls integration', function()
+  local Jsonls
```
we can just do 


```suggestion
  local Jsonls require('codesettings.integrations.jsonls')
```

mrjones2014 commented on spec/jsonls_spec.lua:
```diff
@@ -0,0 +1,222 @@
+      assert.same({ original_schema }, existing_schemas)
+      -- The returned schemas should have more items (original + plugin schemas)
+      assert.is_true(#schemas >= original_length)
+    end)
```
this is not really testing anything IMO

mrjones2014 commented on spec/jsonls_spec.lua:
```diff
@@ -0,0 +1,222 @@
+      -- Assert: The existing schema should be first (preserved)
+      assert.is_not_nil(schemas)
+      assert.is_true(#schemas > 0)
+      assert.same(existing_schema, schemas[1])
```
```suggestion
      local raw_schemas = require('codesettings.build.schemas').get_schemas()
      assert.is_true(#schemas > #raw_schemas)
      assert.same(existing_schema, schemas[1])
```

mrjones2014 commented on spec/jsonls_spec.lua:
```diff
@@ -0,0 +1,222 @@
+      assert.is_not_nil(schemas)
+      assert.is_table(schemas)
+    end)
+  end)
```
these tests feel like we're testing `vim.tbl_get` and `vim.lsp.config`, I think we can just drop them.

mrjones2014 commented on spec/jsonls_spec.lua:
```diff
@@ -0,0 +1,222 @@
+    end)
+  end)
+
+  describe('setup', function()
```
I think we can drop this whole block, the interesting logic all happens in `get_schemas`, and there is no way to clear LSP configs in Neovim in between tests so the purity of the tests is questionable.

mrjones2014 reviewed (changes requested):
Thanks for the contribution. I appreciate the tests, though it looks like they were largely generated by the AI tool (as mentioned), and that unfortunately comes through in the results.

Several of the tests are redundant, don’t assert meaningful behavior, or claim to test one thing while the assertions don't actually test that. It feels like many were added without a clear understanding of what they’re verifying or even a first pass review, which ends up just meaning I have to spend my time reviewing it.

While AI tools can absolutely be helpful in writing code, I’d really appreciate it if you could take the time to review and refine their output before adding it to a PR in the future.

okuuva commented on spec/jsonls_spec.lua:
```diff
@@ -0,0 +1,222 @@
+      assert.is_not.equal(schemas1, schemas2)
+      -- But content should be the same
+      assert.same(schemas1, schemas2)
+    end)
```
Same as above.

okuuva commented:
@mrjones2014 first of all: I'm sorry I didn't check the tests properly. I did skim them through and cleaned up some blatant mistakes but I should've spent more time going through them thoroughly. Sorry for wasting your time 🙇🏼‍♂️

I addressed all feedback. I still think testing the caching is valuable so I moved the tests to their own block and described them better. If you still think they're unnecessary they are now in their separate commit so dropping them is easy enough.

mrjones2014 reviewed (changes requested):
Looking much better now, thanks for cleaning it up! Just a few remaining nits.
